lstchain/calib/camera/time_sampling_correction.py

```diff
@@ -0,0 +1,118 @@
+import h5py
+import numpy as np
+
+
+from ctapipe.core import Component
+from ctapipe.core.traits import Unicode
+
+__all__ = [
+    'TimeSamplingCorrection',
+    ]
+
+
+
+class TimeSamplingCorrection(Component):
+    """
+        The PulseTimeCorrection class to correct time pulse
+        using Fourier series expansion.
+    """
+
+    time_sampling_correction_path = Unicode(
+        '',
+        help='Path to the waveform sampling correction file',
+        allow_none = True,
+    ).tag(config=True)
+
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+        self.time_sampling_coefficients = None
+
+        self.load_sampling_coefficient_file()
+
+    def load_sampling_coefficient_file(self):
+        """
+            Function to load sampling coefficient file.
+        """
+
+        try:
+            with h5py.File(self.time_sampling_correction_path, 'r') as hf:
+                self.time_sampling_coefficients = np.array(hf['sampling_interval_coefficient'])
+        except:
+            self.log.error(f"Problem in reading sampling coefficient file {self.time_sampling_correction_path}")
+
+
+    def get_corrections(self, event, telid):
+        """
+        Get the time/charge sampling corrections for one event and one telescope
+
+        Parameters
+        ----------
+        event: general event container
+        telid: id of the telescope
+
+        Return
+        ------
+        sampling_corrections: np.array (n_gains, n_pixels, n_samples) with the correction factors
+        """
+
+        n_gains = 2
+        n_modules = 265  # number of modules in LST's camera.
+        n_pixel_per_module = 7  # number of pixels per one module.
+        size_drs4 = 1024
+        n_pixels = 1855
+        roi = 36
+
+        # shift the first capacitor to the value used in the r1
+        r1_roi_start = 3
```

Comment:
Is this the same as `LSTR0Corrections.r1_sample_start`? If yes, what happens if someone uses a different value of three for that traitlet? Why is this hard coded here?

Reply:
> Is this the same as `LSTR0Corrections.r1_sample_start`? 

yes

> If yes, what happens if someone uses a different value of three for that traitlet? 

it would be a bad choice ;-)

> Why is this hard coded here?

now we have all these hardcoded values in  the source calibrator. Are there objections to move this code there? (one could be that these corrections are more offline than online)

Reply:
> it would be a bad choice ;-)

If this value should never be different from 3, then we shouldn't make it a traitlet.

Reply:
Let say that this value must be the same that one use for the drs4 calibration

Reply:
So it should be an argument to the function and the caller should give `LSTR0Corrections.r1_sample_start` to make sure this is always correct.